### ISSUE TITLE
[MacOS] Menu example does not show menus

### ISSUE THREAD

**Issue by mikeylemmon** [r0x0r/pywebview#1534]
### Specification
- pywebview version: `master` branch (5.3.2+ [commit f19a758406])
- operating system: macOS 14.6.1 (Sonoma) [M1 Pro]
- web renderer: Cocoa

### Description

Menus don't currently seem to be working on MacOS — running the menu example results only in a "View" menu (shown below) regardless of what is passed to the menu argument:

<img width="307" alt="image" src="https://github.com/user-attachments/assets/7bf9cf38-72e0-4b7d-90d7-a5d042721bff">

This behavior is the same when bundled with py2app.

A tiny bit of digging reveals that the issue can be fixed by commenting out the [`self._clear_main_menu()` call](https://github.com/r0x0r/pywebview/blob/f19a7584/webview/platforms/cocoa.py#L638) that was added to cocoa.py's `first_show` method in #1407, but I'm guessing that removing the call also re-introduce the "ever-growing main menu bug". @lanzz were custom menus working when you first implemented your fix for the menu bug?

### Practicalities

- YES I am willing to work on this issue myself.

- NO I am not prepared to support this issue financially.

**Comment by lanzz** [comment]
I haven't used custom menus myself, the growing main menu bug I encountered during running the tests for an unrelated change. Sounds like that whole reset menu logic there needs to be moved to the same location where the custom menu is applied.

**Comment by r0x0r** [comment]
Yes, @mikeylemmon is correct that `_clear_main_menu` is the culprit. The menu logic needs to be refactored. Maybe window specific menu can be incorporated as part of this refactor as well

**Comment by r0x0r** [comment]
I pushed a fix to master. Please test.

**Comment by mikeylemmon** [comment]
I just tested it out and menus are working, thanks for the fix!
